src/main/java/com/kryptnostic/rhizome/hazelcast/serializers/SetStreamSerializers.java

```diff
@@ -17,6 +20,40 @@
         }
     }
 
+    public static void fastUUIDSetSerialize( ObjectDataOutput out, Set<UUID> object ) throws IOException{
+        long[] least = new long[object.size()];
+        long[] most = new long[object.size()];
+        int i = 0;
+        for ( UUID uuid : object ) {
+            least[i] = uuid.getLeastSignificantBits();
+            most[i] = uuid.getMostSignificantBits();
+            i++;
+        }
+        out.writeInt( i );
+        out.writeLongArray( least );
+        out.writeLongArray( most );
+    }
+    public static OrderedUUIDSet fastOrderedUUIDSetDeserialize( ObjectDataInput in ) throws IOException{
+        int size = in.readInt();
+        OrderedUUIDSet set = new OrderedUUIDSet( size );
+        return (OrderedUUIDSet) processEntries( set, in );
+    }
+
+    public static UUIDSet fastUUIDSetDeserialize( ObjectDataInput in ) throws IOException{
+        int size = in.readInt();
+        UUIDSet set = new UUIDSet( size );
+        return (UUIDSet) processEntries( set, in );
+    }
+
+    private static Set<UUID> processEntries( Set<UUID> set, ObjectDataInput in ) throws IOException {
+        long[] least = in.readLongArray();
+        long[] most = in.readLongArray();
+        for ( int i=0; i < set.size(); i++ ) {
```

Comment:
Process entries needs the size as set.size() will be zero, even though the initial capacity is correct.